pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -609,3 +621,31 @@ func translateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
 
 	return &out
 }
+
+func getHashPolicy(configStore model.IstioConfigStore, hostname model.Hostname) *route.RouteAction_HashPolicy {
+	if configStore == nil {
+		return nil
+	}
+
+	destinationRule := configStore.DestinationRule(hostname)
+	if destinationRule == nil {
+		return nil
+	}
+
+	rule := destinationRule.Spec.(*networking.DestinationRule)
+	consistentHash := rule.GetTrafficPolicy().GetLoadBalancer().GetConsistentHash()
```

Comment:
You need to check if there is a consistent hash/lb setting for the destination subset as well. Look at cluster.go to see how its using the subset to override top level lb settings